core/src/main/scala/org/apache/gearpump/util/Graph.scala

```diff
@@ -256,10 +258,28 @@ object Graph {
     implicit def anyToPath[N, E](any: N): Path[N, E] = Node(any)
   }
 
-  implicit class Node[N, E](any: N) extends Path[N, E](List(Left(any))) {
+  implicit class Node[N, E](self: N) extends Path[N, E](List(Left(self))) {
     
     override def ~[Edge](edge: Edge): Path[N, Edge] = {
-      new Path(List(Left(any), Right(edge)))
+      new Path(List(Left(self), Right(edge)))
+    }
+
+    override def ~>[Node >: N](node: Node): Path[Node, E] = {
+      new NodeList(List(self, node))
+    }
+
+    override def to[Node >: N, Edge >: E](node: Node, edge: Edge): Path[Node, Edge] = {
+      this ~ edge ~> node
+    }
+  }
+
+  class NodeList[N, E](nodes: List[N]) extends Path[N, E](nodes.map(Left(_))) {
+    override def ~[Edge](edge: Edge): Path[N, Edge] = {
+      new Path(nodes.map(Left(_)) :+ Right(edge))
+    }
+
+    override def ~>[Node >: N](node: Node): Path[Node, E] = {
+      new NodeList(nodes :+ node)
```

Comment:
What happens if path is: a ~> b ~RRPartitioner ~> c?

What is the partitioner for a~> b? i assume it shall be HashPartitioner. But, the code seems it will be all RRPartitioner.